libs/ui-react/src/lib/Components/ListItem/ListItem.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { cn } from '@ldls/utils-shared';
+import { IconSize } from '../Icon/Icon';
+import { Tag, type TagProps } from '../Tag/Tag';
+
+export interface ListItemProps
+  extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'children'> {
+  title: string;
+  description?: string;
+  descriptionTagProps?: TagProps;
+  leadingIcon?: React.ComponentType<{ size?: IconSize; className?: string }>;
+  trailingContent?: React.ReactNode;
```

Comment:
while working on the input I wanted to add description to the props so it helps the user understand what is it about when you hover over it, I think we should do this often for props that are atypical like this  `trailingContent`
```suggestion
  /**
   * Custom content to render on the right side of the List-item.
   * @example trailingContent={<Icon />}
   */
  trailingContent?: React.ReactNode;
```
<img width="638" height="249" alt="image" src="https://github.com/user-attachments/assets/6d1bb547-ff74-4d47-9930-71b569cc2800" />